backends/xnnpack/partition/config/xnnpack_config.py

```diff
@@ -174,11 +175,19 @@ def _check_inputs_are_valid_dtypes(self, node, valid_dtypes):
             if arg_val.dtype not in valid_dtypes:
                 return False
 
+            # Check for mixed dtypes
+            if reference_dtype is None:
+                reference_dtype = arg_val.dtype
+            elif arg_val.dtype != reference_dtype:
+                return False
+
         return True
 
     def _check_outputs_are_valid_dtypes(self, node, valid_dtypes):
-        # Check outputs are valid dtype
+        # Check outputs are valid and have the same dtypes
```

Comment:
i'm hesistant to add the output check, because for ops like max_pooling they return float values and int64 indicies.